sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { getUniqueName, isBrowser, isRecording, isPlayingBack } from "./utils";
+import { NiseRecorder, NockRecorder, BaseRecorder, setEnviromentOnLoad } from "./baseRecorder";
+
+/**
+ * @export
+ * An interface that allows recording and playback capabilities for the tests in Azure TS SDKs.
+ */
+export interface Recorder {
```

Comment:
I find it slightly confusing that the interface `Recorder` interface is not really for the real recorders like the `NockRecorder` but is for a set of functions.

Reply:
`NockRecorder` is not exposed to the users and is internal to the recorder package.

If you still feel like it might be confusing, `Recorder` can be renamed to `RecorderHandler` to differentiate it from `NockRecorder`/`BaseRecorder`.

What say?

Reply:
If only one `Recorder` type is exposed, I think it's fine to keep the name.